lib/navigation/view/web_shell.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:acs_upb_mobile/navigation/model/navigation_state.dart';
+import 'package:acs_upb_mobile/navigation/service/app_navigator.dart';
+import 'package:acs_upb_mobile/navigation/service/app_router_delegates.dart';
+import 'package:acs_upb_mobile/pages/home/home_page.dart';
+import 'package:acs_upb_mobile/pages/people/view/people_page.dart';
+import 'package:acs_upb_mobile/pages/portal/view/portal_page.dart';
+import 'package:acs_upb_mobile/pages/timetable/view/timetable_page.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_feather_icons/flutter_feather_icons.dart';
+
+import 'dummy_search_bar.dart';
+
+class WebShell extends StatefulWidget {
+  WebShell({Key key, this.navigationState}) : super(key: key);
+
+  final NavigationStateProvider navigationState;
+  final PageStorageBucket bucket = PageStorageBucket();
+
+  @override
+  _WebShellState createState() => _WebShellState();
+}
+
+class _WebShellState extends State<WebShell> {
+  InnerRouterDelegate _innerRouterDelegate;
+  ChildBackButtonDispatcher _backButtonDispatcher;
+
+  @override
+  void initState() {
+    super.initState();
+    _innerRouterDelegate =
+        InnerRouterDelegate(navigationState: widget.navigationState);
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _backButtonDispatcher = Router.of(context)
+        .backButtonDispatcher
+        .createChildBackButtonDispatcher();
+  }
+
+  @override
+  void dispose() {
+    _innerRouterDelegate.dispose();
+    super.dispose();
+  }
+
+  void handleNavigationBar(int index) {
+    setState(() {
+      widget.navigationState.selectedTab = index;
+
+      switch (index) {
+        case 0:
+          AppNavigator.pushNamed(context, HomePage.routeName);
+          break;
+        case 1:
+          AppNavigator.push(
+            context,
+            MaterialPageRoute<Widget>(
+              builder: (_) => const Center(child: Text('ok')),
+            ),
+            webPath: '${TimetablePage.routeName}',
+          );
+          // AppNavigator.pushNamed(context, TimetablePage.routeName);
+          break;
+        case 2:
+          AppNavigator.pushNamed(context, PortalPage.routeName);
+          break;
+        case 3:
```

Comment:
The tabIndex-page widget correspondence is repeated many times throughout the code. This is not ideal - God forbid you'd wanna add another tab or change the ordering, you'd have to change:
- main.dart
- app_navigator.dart
- app_router_delegates.dart
- web_shell.dart

And I may well have missed a usage. We should have some sort of map from tabIndex to widget and use that everywhere.

Reply:
https://github.com/student-hub/acs-upb-mobile/pull/265/commits/056622f6145623c524a8716e98e3eafa8c332c40 this commit solves this issue.
`main.dart` was not modified because the case [here](https://github.com/student-hub/acs-upb-mobile/blob/056622f6145623c524a8716e98e3eafa8c332c40/lib/main.dart#L156-L159) is related to mobile usage.